Don't append standard ports to the request header.

capnm opened the pull request:
Breaks the SSL communication with some servers,
follow the curl, wget, firefox & co

Fixes #9146

bojidar-bg commented:
@capnm Just to make sure, did you check if it runs fine with this change?

bojidar-bg commented on core/io/http_client.cpp:
```diff
@@ -96,7 +96,12 @@ Error HTTPClient::request_raw(Method p_method, const String &p_url, const Vector
 
 	String request = String(_methods[p_method]) + " " + p_url + " HTTP/1.1\r\n";
-	request += "Host: " + conn_host + ":" + itos(conn_port) + "\r\n";
+	if (ssl && conn_port == 443) {
```
I think you should also not add port 80 in case it is non-ssl.
Something like: (untested)
```c++
if ((ssl && conn_port == 443) || (!ssl && conn_port == 80)) {
	…
```
(applies to other if below as well)

capnm commented:
> @capnm Just to make sure, did you check if it runs fine with this change?

Sure, the SSL connection works now  :sunglasses:
 
```
request: 'https://www.capitanio.org/tests/godot/' = 0
CERT STR: /CN=www.capitanio.org
VALID: 1
CONNECTION RESULT: 1
cert_ok: 1
XXXX SSL_write [122]
-=>GET /tests/godot/ HTTP/1.1
Host: www.capitanio.org
User-Agent: GodotEngine/3.0.alpha.custom_build (X11)
Accept: */*

<=-
result: 0
response_code: 200
headers: [Date: Tue, 13 Jun 2017 17:16:53 GMT, Server: Apache/2.2.31 (Unix), X-Powered-By: PHP/7.0.17, Cache-Control: max-age=60, Expires: Tue, 13 Jun 2017 17:17:53 GMT, Vary: Accept-Encoding, Content-Type: text/html, Transfer-Encoding: chunked]
body.length: 150
body:
<!DOCTYPE html>
<html>
<head>
<title>gogot https test</title>
<meta charset="utf-8" />
</head>
<body>
	<h1>Waiting for godot 3.1</h1>
</body>
</html>
```

capnm commented:
> I think you should also not add port 80 in case it is non-ssl.

Ok, looks like it's also not appended by the other clients:

```
curl -vv http://capitanio.de:80/tests/godot/
*   Trying 81.169.145.149...
* Connected to capitanio.de (81.169.145.149) port 80 (#0)
> GET /tests/godot/ HTTP/1.1
> Host: capitanio.de
> User-Agent: curl/7.47.0
> Accept: */*
```

capnm committed:
Don't append standard ports to the request header.

Breaks the SSL communication with some servers,
do the same that the other curl, wget, firefox & co clients do.

Fixes #9146

capnm commented:
@bojidar-bg The godot http implementation has IMO generally a many subtle bugs, but it is ok, if you can control the server content. I've tested some random urls and everything seems to work atl least better as before :-)

Everything bellow ended with response_code: 200 + body:
```
request: 'http://www.capitanio.de:80/tests/godot/' = 0
GET /tests/godot/ HTTP/1.1
Host: www.capitanio.de

request: 'http://www.capitanio.de/tests/godot/' = 0
GET /tests/godot/ HTTP/1.1
Host: www.capitanio.de

request: 'http://www.capitanio.de:21/tests/godot/' = 0
GET /tests/godot/ HTTP/1.1
Host: www.capitanio.de:21

request: 'http://rpi.local:8888/doku.php' = 0
GET /doku.php HTTP/1.1
Host: rpi.local:8888

-----
request: 'http://google.com' = 0
GET / HTTP/1.1
Host: google.com

request: 'https://google.com' = 0
Host: google.com
XXXX SSL_write
-=-GET / HTTP/1.1
Host: google.com

----

request: 'https://github.com' = 0
Host: github.com
XXXX SSL_write
-=-GET / HTTP/1.1
Host: github.com

request: 'http://github.com:80' = 0
Host: github.com
````

capnm commented:
@bojidar-bg This should be also cherry-picked to the 2.x  branch.

bojidar-bg reviewed (approved):
LGTM

capnm commented:
@akien-mga could you cherrypick it to the 2.x branch as well?

bojidar-bg commented:
@capnm I think he would do it when he starts getting 2.1.4 ready anyway.
